### PR TITLE
Add color palette and design guidelines

### PR DESCRIPTION
**Pull request by upsj**
Adds a design guideline document with concrete hex values and usage suggestions (based on how the colors are used on ginkgo-project.github.io)

### CODE REVIEW

**Review by tcojean — APPROVED** [review]

**Review by pratikvn — COMMENTED** [review]
LGTM! I am not sure all the colors match. I am referring to the [svg logo](https://github.com/ginkgo-project/ginkgo/edit/develop/assets/logo_dark_horizontal.svg), where I see some hex codes that are different. 

Also, can we add the font we use for the logo ? I am not sure which font that is to be honest.

**Comment by upsj** [comment]
I used the colors @hartwiganzt sent me, plus one from ginkgo-project.github.io that was missing in the list. Font is a great idea!

**Comment by upsj** [comment]
Merryweather

File: assets/design.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+The colors are, from left to right
+1. ginkgo-orange-1 `#F7A71E`, suggested use: highlight color
+2. ginkgo-orange-2 `#FCB725`, suggested use: small elements like buttons
+3. ginkgo-orange-3 `#FFCB05`, suggested use: background
+4. ginkgo-gray-1 `#6D6E71`
+5. ginkgo-gray-2 `#939598`
+6. ginkgo-gray-3 `#D1D3D4`
+
+The font used in the logo is Merryweather Black.
```
Comment (pratikvn) [inline assets/design.md:15]: :)
```suggestion
The font used in the logo is Merriweather Black.
```

**Review by pratikvn — APPROVED** [review]
LGTM! Thanks for figuring out the font.

**Pull request merged**